plugins/modules/rudder_node_settings.py

```diff
@@ -280,6 +280,44 @@ def _send_request(
                     status_code, resp.content
                 ),
             )
+            
+    def check_NodeSettingValue(self, node_id, property_name, property_value):
+        """Function to check if a setting or property as already set on node via the API
+        
+        Args:
+            values (str): Values to check in API
+            property_name (str): Name of your node property to check
+            property_value (str): Value of your node property to check
+        
+        Returns:
+            bool: Return if yes (true) or no (false) a value as already set (default: False)
+        """
+        url = "/api/latest/nodes/{node_id}\?include=default".format(node_id=self.node_id)
+
+        if property_name is not None and property_value is not None:
+            response = self._send_request(
+                path=url,
+                headers=self.headers,
+                serialize_json=False,
+                method="GET",
+            )
+            
+            binary = response.content
+            output = json.loads(binary)
+            
+            path = output['data']['nodes'][0]['properties']
+
+            for out in path:
```

Comment:
You want to test the whole properties at once, not one key per function call. You will have to inverse the logic here and iterate over the expected properties, and check that each of them is present in the current properties with current value.